### DISCUSSION TITLE
Should murex change its name?

### DISCUSSION THREAD

**Discussion by lmorg** [lmorg/murex#633] [Ideas]
This is something I've been increasingly pondering lately. The issue being that the term `murex shell` is abysmal for SEO.

I still like the idea behind the name, so if I do rename it I'd probably go with something like `mxshell` or `mxsh`. Thus keeping the spirit of the original name alive.

If I do rename then what do I do with the existing assets:
- domain: https://murex.rocks -- presumably that should be changed too
- binary `murex` -- changed also? And to what, `mxsh`?
- this git repository?

I'm not even sure if renaming this project is a smart idea. But if I am going to do this then it should probably happen sooner rather than later.

What are other peoples thoughts?

**Comment by lmorg** [comment]
...or maybe `uxshell` since UX is a primary concern and following on from a suggestion made by @kmatt in https://github.com/lmorg/murex/issues/437#issuecomment-1250487985

**Comment by kmatt** [comment]
To be clear I was not suggesting a project name change, just attempting some logo art. Personally I like the name Murex, as uxshell is too close to things like nushell.

**Comment by verkaro** [comment]
How about the Murex Interactive Shell or the Mission Impossible Shell (MISH) which will inadvertantly be spoken mish-shell? Like a charming girl I once knew whose surprising speech impediment was luckily confined to one word -- yes, ie. yesh.

Mish Shell has current SEO presence as a singer but pop singers rise and fall in relevance while Murex has long term relevance and will likely rise to the top of search results and stay there.

Whatever name you decide you'd want to keep murex.rocks alive at the very least to redirect.

BTW. The murex shell does rock. I am not a coder. I am a small business owner who prefers to work from a cli as much as possible -- which with modern improvements is becoming a vi-able choice.

**Comment by lmorg** [comment]
@verkaro 

> How about the Murex Interactive Shell or the Mission Impossible Shell (MISH) which will inadvertantly be spoken mish-shell? Like a charming girl I once knew whose surprising speech impediment was luckily confined to one word -- yes, ie. yesh.
>
> Mish Shell has current SEO presence as a singer but pop singers rise and fall in relevance while Murex has long term relevance and will likely rise to the top of search results and stay there.

Interesting suggestion. I think MISH would be too similar to Mosh (https://mosh.org/) but I think MI Shell has more potential. Could still be pronounced "Michelle".

> Whatever name you decide you'd want to keep murex.rocks alive at the very least to redirect.

As a redirect, I agree.

> BTW. The murex shell does rock. I am not a coder. I am a small business owner who prefers to work from a cli as much as possible -- which with modern improvements is becoming a vi-able choice.

Thank you very much. It's always great to hear feedback.

**Comment by jmcomets** [comment]
FWIW, I used to work at a company named Murex. It's a Paris-based software company in the financial sector, and has a couple internal binaries with names similar to those mentioned in here (`mx`, `mx.sh`, `mxsh`, ...).

EDIT: Adding some links here for more context

[Wikipedia page](https://en.wikipedia.org/wiki/Murex_(financial_software))
[Company page](https://www.murex.com/)

**Comment by distek** [comment]
@lmorg do you have a preference on what line to go down? Like, would you prefer to keep it as a reference to a shell adorning (sea) creature? Or would something like `smash` work just because it sounds fun?

**Comment by orefalo** [comment]
IMHO - No

fish... murex..  it's deep.. really deep ;-)

However... it needs better branding.

Anyone in this community with skills to redo a pretty website?
Looks like all the shells are stuck with a design from the 70s - text based ;-/

**Comment by orefalo** [comment]
I found this documentation framework for your review - https://alexxnb.github.io/svelte-docs/introduction
  I know how developers are sticky with tech - it's not built in go, but rather in nodejs.
  I found the capabilities & CX interesting.

The idea would be to redo the complete site with it, or course with a different theme.
  on this topic, I also found a logo -> https://www.logodesign.net/image/celtic-letter-m-5100ld

Let me know your thoughts,

**Comment by lmorg** [comment]
I agree the current site isn't pretty, but it's just static HTML and a little CSS. There isn't even any Javascript. The rational behind that is the content on the site is identical to the markdown documents included in the git repository (some OS packages of Murex do install the markdown docs along with the executable) and the same as the docs that come up with type `help command`.

While I'm happy for a site redesign, I would want the website to still be "powered" by the markdown docs. Ideally not rendering the HTML in the browser too. The fact that this site works with JS turned off I think is a feature as it means it can work with terminal based browsers too.

The other thing I like about this is that every merge `master` I do with the shell automatically updates the docs on the website.

Maybe I've been over thinking / overly restrictive with my requirements. But that was the rational behind the current design.

---

The architecture is basically just using `pandoc` and injecting a header and footer html file:
1. https://github.com/lmorg/murex/tree/master/gen/website (header / footer)
2. https://github.com/lmorg/murex/blob/master/test/ci-website.sh (website generation)
3. https://github.com/lmorg/murex/blob/master/gen/website/find-exec.mx (html page generation)

The whole thing is hosted on AWS S3 with Route 53 for DNS and CloudFront as a CDN (not that I even need CloudFront tbh) and costs me something like $2 a month.

**Comment by orefalo** [comment]
Hi, I will resume discussions on this threat.

First let me reshare the the link  - https://orefalo.github.io/murex-docs/

Key Benefits

- free hosting with your own domain name (that's something you will need to configure at some point when you bring the project under your ownership)
- based on vuepress (no need to learn it, just edit .md files)
- theme based on [vuepress-theme-hope](https://theme-hope.vuejs.press/) - which comes with tons of features, check it out!
- auto generates robots.txt and sitemap
- better landing page
- local search index
- nice & slick Murex theme!
- auto publishing upon content change
- content comments

I am trying to get the following OKs and feedback:

1. color there: green ? feel free to select another color, but let's do it early
2. logo?  I've done those in a few clicks, I personally like the ">>_" which is characteristic to Murex.
3. Content, I am slowing moving content into the new viewpress structure.
  along with it comes additional capabilities - for instance diagrams, pictures, callouts, icons..etc

Sincerely,

**Reply by orefalo** [reply]
in complete honesty, (I will criticize my own work)
I like the features of vuepress and the layout.

there is something off with the theme/colors.

**Comment by lmorg** [comment]
thanks for the work on this. I think this deserves it's own discussion so created https://github.com/lmorg/murex/discussions/646
